databuilder/extractor/dashboard/tableau/tableau_dashboard_owner_extractor.py

```diff
@@ -0,0 +1,102 @@
+from databuilder.transformer.dict_to_model import DictToModel, MODEL_CLASS
+from databuilder.transformer.template_variable_substitution_transformer import \
+    TemplateVariableSubstitutionTransformer, TEMPLATE, FIELD_NAME as VAR_FIELD_NAME
+from databuilder.transformer.timestamp_string_to_epoch import TimestampStringToEpoch, FIELD_NAME
+
+LOGGER = logging.getLogger(__name__)
+
+
+class TableauDashboardOwnerExtractor(Extractor):
+    """
+    """
+
+    def init(self, conf):
+        # type: (ConfigTree) -> None
+
+        self._conf = conf
+        self._auth = TableauDashboardAuth(self._conf)
+        self.query = """query {
+          workbooks {
+            name
+            projectName
+          }
+        }"""        
+        self._extractor = self._build_extractor()
+
+        transformers = []
+        dict_to_model_transformer = DictToModel()
+        dict_to_model_transformer.init(
+            conf=Scoped.get_scoped_conf(self._conf, dict_to_model_transformer.get_scope()).with_fallback(
+                ConfigFactory.from_dict(
+                    {MODEL_CLASS: 'databuilder.models.dashboard.dashboard_owner.DashboardOwner'})))
+        transformers.append(dict_to_model_transformer)
+        self._transformer = ChainedTransformer(transformers=transformers)
+
+
+    def extract(self):
+        # type: () -> Any
+
+        record = self._extractor.extract()
+        if not record:
+            return None
+
+        return self._transformer.transform(record=record)
+
+    def get_scope(self):
+        # type: () -> str
+
+        return 'extractor.tableau'
+
+    def _build_extractor(self):
+        """
+        """
+        extractor = TableauGraphQLApiOwnerExtractor()
+        tableau_extractor_conf = \
+            Scoped.get_scoped_conf(self._conf, extractor.get_scope())\
+                  .with_fallback(self._conf)\
+                  .with_fallback(ConfigFactory.from_dict({
+                                                          STATIC_RECORD_DICT: {'product': 'tableau'}
+                                                          }
+                                                         )
+                                 )
+        extractor.init(conf=tableau_extractor_conf, auth_token=self._auth.token, query=self.query)
+        return extractor
+
+class TableauGraphQLApiOwnerExtractor(TableauGraphQLApiExtractor):
+    """docstring for TableauDashboardMetadataExtractor"""
+    def execute(self):
+        response = self.execute_query()
+
+        for workbook in [workbook for workbook in response['workbooks'] if workbook['projectName'] not in ["ZZZ - Archived", "WIP", "Tableau Samples"]]:
+            data = {}
+            data['dashboard_group_id'] = workbook['projectName']
+            data['dashboard_id'] = html.escape(str(workbook['name']))
+            email_username = re.sub(" ", "-", data['dashboard_group_id']).lower()
+            data['email'] = email_username + "-team@gusto.com"
```

Comment:
I think we'd have to come up with some static mapping of top-level Tableau projects -> teams/individuals w/in Gusto. There is an "owner" for each workbook, but IIRC Jeff mentioned that that was not a useful indicator, since when someone re-publishes a workbook it then tags them as the new "owner" even if they did not really change anything. However I've slept since then and I may have misunderstood initially, so I'll follow up with BI and clarify.